Subject: Avoid error dialog caused by symbol lookup error
From: Lubos Lunak
Bug: bnc#444800
Patch-upstream: no

For whatever reason launching kdm control module leads to a symbol lookup error
during its exit, which leads to its exit code being 127, i.e. not found.
Add ugly hack to hide this case.

--- kdesu/kdesu/kdesu.cpp.sav	2010-03-16 10:33:25.000000000 +0100
+++ kdesu/kdesu/kdesu.cpp	2010-05-28 16:41:00.000000000 +0200
@@ -71,6 +71,8 @@ QByteArray dcopNetworkId()
 
 static int startApp();
 
+static bool commandWasOk = false;
+
 int main(int argc, char *argv[])
 {
     // FIXME: this can be considered a poor man's solution, as it's not
@@ -135,7 +137,7 @@ int main(int argc, char *argv[])
 
     int result = startApp();
 
-    if (result == 127)
+    if (result == 127 && !commandWasOk)
     {
         KMessageBox::sorry(0, i18n("Command '%1' not found.", QString::fromLocal8Bit(command)));
     }
@@ -258,6 +260,8 @@ static int startApp()
             command += QFile::encodeName(arg);
         }
     }
+    QStringList commandItems = QString::fromLocal8Bit( command ).split( ' ' );
+    commandWasOk = ( !commandItems.isEmpty() && !KStandardDirs::findExe( commandItems.first()).isEmpty());
 
     // Don't change uid if we're don't need to.
     if (!change_uid)
